employee-rostering-benchmark/src/main/java/org/optaweb/employeerostering/OptaWebEmployeeRosteringBenchmarkApplication.java

```diff
@@ -14,45 +14,56 @@
  * limitations under the License.
  */
 
-package org.optaweb.employeerostering.benchmark;
+package org.optaweb.employeerostering;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import javax.persistence.EntityManager;
-import javax.persistence.EntityManagerFactory;
-import javax.persistence.Persistence;
+import javax.persistence.PersistenceContext;
 
 import org.optaplanner.benchmark.api.PlannerBenchmark;
 import org.optaplanner.benchmark.api.PlannerBenchmarkFactory;
-import org.optaplanner.core.api.solver.SolverFactory;
 import org.optaweb.employeerostering.domain.roster.Roster;
 import org.optaweb.employeerostering.service.roster.RosterGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.transaction.annotation.Transactional;
 
-public class OptaWebEmployeeRosteringBenchmarkApp {
+@SpringBootApplication
+public class OptaWebEmployeeRosteringBenchmarkApplication implements ApplicationRunner {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
 
     public static void main(String[] args) {
+        SpringApplication.run(OptaWebEmployeeRosteringBenchmarkApplication.class, args);
+    }
+
+    @Override
+    @Transactional
+    public void run(ApplicationArguments args) {
         List<Roster> rosterList = generateRosters();
 
-        SolverFactory<Roster> solverFactory = SolverFactory.createFromXmlResource(
-                "org/optaweb/employeerostering/service/solver/employeeRosteringSolverConfig.xml");
-        PlannerBenchmarkFactory benchmarkFactory = PlannerBenchmarkFactory.createFromSolverFactory(solverFactory);
+        PlannerBenchmarkFactory benchmarkFactory = PlannerBenchmarkFactory.createFromXmlResource(
+                "employeeRosteringBenchmarkConfig.xml", getClass().getClassLoader());
         PlannerBenchmark plannerBenchmark = benchmarkFactory.buildPlannerBenchmark(rosterList);
-        plannerBenchmark.benchmark();
+        plannerBenchmark.benchmarkAndShowReportInBrowser();
```

Comment:
Not 100 % sure, but isn't there a method that wouldn't try to open the browser? Somehow opening a browser in a Spring app doesn't feel right to me.

Reply:
Yes, the `benchmark()` method doesn't try to open the browser. You're probably right, it does feel strange, I'll change it back.